packages/accordion/src/useAccordion.ts

```diff
@@ -5,185 +5,144 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import { useState, HTMLProps, useMemo } from 'react';
-import { useUIDSeed } from 'react-uid';
+import { useState, useMemo, useCallback } from 'react';
 import {
-  composeEventHandlers,
   KEY_CODES,
-  getControlledValue
+  composeEventHandlers,
+  getControlledValue,
+  useId
 } from '@zendeskgarden/container-utilities';
 
-export interface IUseAccordionProps {
-  /** Prefixes IDs for the accordion trigger and panels  */
-  idPrefix?: string;
-  /** Sets the expanded sections in a controlled accordion */
-  expandedSections?: number[];
-  /** Sets the default expanded sections in a uncontrolled accordion */
-  defaultExpandedSections?: number[];
-  /** Handles accordion expansion changes */
-  onChange?: (expanded: number) => any;
-  /** Determines if multiple panels can be expanded at the same time in an uncontrolled accordion */
-  expandable?: boolean;
-  /** Determines if panels can be collapsed in an uncontrolled accordion */
-  collapsible?: boolean;
-}
-
-interface IHeaderProps extends HTMLProps<any> {
-  ariaLevel?: number | null;
-  role?: any;
-}
-
-interface ITriggerProps extends HTMLProps<any> {
-  index?: number;
-  role?: any;
-  tabIndex?: any;
-}
-
-interface IPanelProps extends HTMLProps<any> {
-  index?: number;
-  role?: any;
-}
-
-export interface IUseAccordionPropGetters {
-  getHeaderProps: <T>(options?: T & IHeaderProps) => any;
-  getTriggerProps: <T>(options?: T & ITriggerProps) => any;
-  getPanelProps: <T>(options?: T & IPanelProps) => any;
-}
-
-export interface IUseAccordionReturnValue extends IUseAccordionPropGetters {
-  expandedSections: number[];
-  disabledSections: number[];
-}
+import { IUseAccordionProps, IUseAccordionReturnValue } from './types';
 
-export function useAccordion({
+export function useAccordion<Value>({
   idPrefix,
+  sections,
   expandedSections,
-  onChange,
+  defaultExpandedSections,
+  onChange = () => undefined,
   expandable = true,
-  collapsible = true,
-  defaultExpandedSections
-}: IUseAccordionProps = {}): IUseAccordionReturnValue {
-  const isControlled = expandedSections !== null && expandedSections !== undefined;
-  const seed = useUIDSeed();
-  const prefix = useMemo<string>(
-    () => idPrefix || seed(`accordion_${PACKAGE_VERSION}`),
-    [idPrefix, seed]
-  );
+  collapsible = true
+}: IUseAccordionProps<Value>): IUseAccordionReturnValue<Value> {
+  const prefix = useId(idPrefix);
   const TRIGGER_ID = `${prefix}--trigger`;
   const PANEL_ID = `${prefix}--panel`;
-  const [expandedState, setExpandedState] = useState<number[]>(defaultExpandedSections || [0]);
-
-  const controlledExpandedState = getControlledValue(expandedSections, expandedState)!;
 
+  const isControlled = expandedSections !== null && expandedSections !== undefined;
+  const [expandedState, setExpandedState] = useState<Value[]>(
+    defaultExpandedSections || expandedSections || sections.slice(0, 1)
```

Comment:
Why is this falling back to an expansion state on the first section? Feels like we should be accepting `undefined` as a valid state.

Reply:
Could you explain more on this? `expandedState` does have dependents (like `disabledState`, etc) which does need it to be an Array in certain conditions.

Reply:
Not strictly related to JZ's comment, but I would expect `expandedSections` to come before `defaultExpandedSections` - controlled value takes priority, and the default as the fallback, if given. 

Getting back to JZ's comment, I'm curious on the thinking there as well. Not opposed to it, but keeping the type consistent does help with the internal logic having a consistent type from the get go. I could also see `defaultExpandedSections` having a default arg value of `[]` to make it simpler.